src/main/java/org/mockito/internal/util/Primitives.java

```diff
@@ -60,6 +63,17 @@ public static boolean isAssignableFromWrapper(Class<?> valueClass, Class<?> refe
         return (T) PRIMITIVE_OR_WRAPPER_DEFAULT_VALUES.get(primitiveOrWrapperType);
     }
 
+    /**
+     * Returns the boxed default value for the lambda return type (handling primitive and primitive wrapper types).
+     *
+     * @param consumer the consumer function
+     * @param <T> the function's return type
+     * @return the boxed default value for the lambda return type (handling primitive and primitive wrapper types)
+     */
+    public static <T> T defaultValueForConsumerLambda(Consumer<T> consumer) {
+        Class<?>[] typeArgs = TypeResolver.resolveRawArguments(Consumer.class, consumer.getClass());
```

Comment:
> The return type of Consumer<T> AFAIK cannot be determined in a simpler way.

Maybe the existing [GenericTypeExtractor](https://github.com/mockito/mockito/blob/release/2.x/src/main/java/org/mockito/internal/util/reflection/GenericTypeExtractor.java) works.

```java
Class<?>[] typeArgs =GenericTypeExtractor.genericTypeOf(consumer.getClass(), Object.class, Consumer.class)
```